modules/git/git.go

```diff
@@ -150,11 +150,11 @@ func Init(ctx context.Context) error {
 }
 
 // Fsck verifies the connectivity and validity of the objects in the database
-func Fsck(repoPath string, timeout time.Duration, args ...string) error {
+func Fsck(ctx context.Context, repoPath string, timeout time.Duration, args ...string) error {
 	// Make sure timeout makes sense.
 	if timeout <= 0 {
 		timeout = -1
 	}
-	_, err := NewCommand("fsck").AddArguments(args...).RunInDirTimeout(timeout, repoPath)
+	_, err := NewCommand("fsck").SetParentContext(ctx).AddArguments(args...).RunInDirTimeout(timeout, repoPath)
```

Comment:
How about to add a method named `NewCommandContext(ctx, "fsck")`?

Reply:
Done.